bin/main

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env ruby
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    entry = is_a?(Range) ? to_a : self
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    entry
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    entry = is_a?(Range) ? to_a : self
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    entry
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    i = 0
+    arr = []
+    while i < size
+      arr << self[i] if yield(self[i])
+      i += 1
+    end
+    arr
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |e| return false unless yield(e) }
+    end
+    if arg.nil?
+      my_each { |e| return false unless e }
+      return true
+    end
+    my_each { |e| return false unless check_validity(e, arg) }
+    true
+  end
+
+  def my_any?(arg = nil, &proc)
+    if block_given?
+      my_each { |e| return true if proc.nil? ? proc.call(e) : yield(e) }
+    else
+      my_each { |e| return true if arg.nil? ? elem : check_validity(e, arg) }
+    end
+    false
+  end
+
+  def my_none?(pattern = nil, &proc)
+    !my_any?(pattern, &proc)
+  end
```

Comment:
- It failed the test when no block or argument is given.
enumerables my_none? when no block or argument is given returns true only if none of the collection members is true
     Failure/Error: expect(false_array.my_none?).to be true

- It also failed the test with regex expressions.     
enumerables my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex.